functions/src/modules/SQR.functions.ts

```diff
@@ -20,52 +20,58 @@ export enum ISoundQualityReportSheet {
-//
-//      2. Send an email to the assignee to notify them of the new allotments
-//              Function --> sendEmailOnNewAllotment
-/////////////////////////////////////////////////
-export const updateFilesOnNewAllotment = functions.database
-  .ref('/sqr/allotments/{allotment_id}')
-  .onCreate((snapshot, context) => {
-    const allotment = snapshot.val();
-    let newDocKey = snapshot.key;
-
-    // loop through the FILES array in the NEW ALLOTMENT object
-    // and update their corresponding file objects
-    allotment.files.forEach(async file => {
-      let sqrRef = db.ref(
-        `/files/${allotment.list}/${file}/soundQualityReporting`
-      );
 
-      let sqrError = await sqrRef.update({
-        status: 'Given',
-        assignee: allotment.assignee,
-        timestampGiven: Math.round(new Date().getTime() / 1000),
-        timestampDone: null,
-      });
+/**
+ * Process registration of webform data
+ * 1. Registers user by transferring data into /users/ path
+ * 2. Check if users already exists
+ * 
+ * @function processRegistration()
+ */
+export const processRegistration = functions.database
+.ref('/webforms/registration/{registration_id}')
+.onCreate(async (snapshot, context) => {
+  const user = snapshot.val();
 
-      if (sqrError == undefined) {
-        // if Successful FILE Update, update the ALLOTMENT accordingly
+  const olduser = await db.ref("/users/").orderByChild("/emailAddress")
+                  .equalTo(user.email_address).once("value");
 
-        // case 1 -- the allotmnet is read from the spreadsheet
-        if (Object.keys(allotment).indexOf('sendNotificationEmail') > -1)
-          db.ref(`/sqr/allotments/${newDocKey}`).update({
-            filesAlloted: true,
-          });
-        // case 2 -- the allotmnet is inputted manually
-        else
-          db.ref(`/sqr/allotments/${newDocKey}`).update({
-            filesAlloted: true,
-            sendNotificationEmail: true,
-          });
-      }
-    })
+  // If the user already exists stop the execution and log the error
+  if (olduser.val() !== undefined) {
+    console.log(`Error! User already exist with email: ${user.email_address}`)
+    return false;
   }
-)
+
+  // if user email doesn't exists prepare the new_user Object
+  const new_user: any = {
+    timestamp: moment.format('x'),
+    emailAddress: user.email_address,
+    phoneNumber: user.contact_number,
+    isAvailableOnWhatsApp: user.i_am_available_on_whatsapp,
+    location: user.country,
+    name: user.your_name,
+    services: (<any>Object).values(user.seva).join(", "),
+    experience: user.experience,
+    influencer: user.where_did_u_hear_about_this_seva,
+    recommendedBy: user.recommended_by,
+    status: user.status,
+  }
+
+  /**
+   * Assigning user roles to new_user 
+   * Get the first letters of service name and assign it to @var new_user's roles
+   */
+  user.services.map(name => name.split(' ').map(piece => piece[0]).join(''))
```

Comment:
I believe this can be done inline in the object construction above… However if it is not easy to do, no need wasting time.

The only thing is that there is no `user.services` property.

Reply:
You're right its not `user.services`. And it can't be done up in the Object that's why i did that separately.